landingai/vision_pipeline.py

```diff
@@ -212,7 +214,7 @@ def show_image(
         """
         # TODO: Should show be a end leaf?
         # Check if we are on a notebook context
-        try:
+        if in_notebook():
```

Comment:
Consider our notebook code is just educational, the notebook code path doesn't get to used in real production environment. do we want to add notebook support to vision pipeline? and in general, do we want to treat notebook support as a first-class citizen in the library?
i'm concerned that the code will be a bit more messy, harder to test, and incurs performance overhead.

alternative solution would be to move the notebook code out of the vision pipeline, and create a dedicated python module for notebook utilities.
i.e. 
```python
from landingai.notebook_utils import display
frame_set = your_vision_pipeline_code...
display(frame_set)
```

Reply:
To me nootebooks are 1st class citizens since they are tool to help reduce customer deployment friction...let's discuss on our daily